src/util/integerfactoring.h

```diff
@@ -0,0 +1,241 @@
+/* Include Guards */
+#ifndef INTEGERFACTORING_H
+#define INTEGERFACTORING_H
+
+
+/* Includes */
+#include <stdio.h>
+#include <stdint.h>
+#include <stdint-gcc.h>
+
+
+/* Defines */
+
+
+
+/* C++ Extern "C" Guard */
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+
+/* Data Structure Prototypes & Typedefs */
+struct GA_FACTOR_LIST;
+typedef struct GA_FACTOR_LIST GA_FACTOR_LIST;
```

Comment:
It's allowed in C but breaks on C++ compilers (and the headers sometimes go though one).  Also this is the style that the rest of the code uses.